cmd/thanos/sidecar.go

```diff
@@ -229,6 +229,16 @@ func runSidecar(
 		})
 	}
 
+	err := addShipper(logger, reg, g, metadata.SidecarSource, m, dataDir, objStoreConfig, uploadCompacted)
+	if err != nil {
+		return errors.Wrap(err, "failed to add block shipper to run group")
+	}
+
+	level.Info(logger).Log("msg", "starting sidecar", "peer", peer.Name())
+	return nil
+}
+
+func addShipper(logger log.Logger, reg prometheus.Registerer, g *run.Group, sourceType metadata.SourceType, m *promMetadata, dataDir string, objStoreConfig *pathOrContent, uploadCompacted bool) error {
 	confContentYaml, err := objStoreConfig.Content()
```

Comment:
I am not a fan of this. You save a line but `addShipper` means to me your are actually adding this NOT adding only IF something happens, can we rearrange? Sorry for being picky ): I don't like methods that hides and suprises.

Reply:
Agreed.